pkg/block/s3/adapter.go

```diff
@@ -849,6 +849,38 @@ func (a *Adapter) ListParts(ctx context.Context, obj block.ObjectPointer, upload
 	return &partsResp, nil
 }
 
+func (a *Adapter) ListMultipartUploads(ctx context.Context, obj block.ObjectPointer) (*block.ListMultipartUploadsResponse, error) {
+	var err error
+	defer reportMetrics("ListMultipartUploads", time.Now(), nil, &err)
+	bucket, key, qualifiedKey, err := a.extractParamsFromObj(obj)
+	if err != nil {
+		return nil, err
+	}
+	input := &s3.ListMultipartUploadsInput{
+		Bucket: aws.String(bucket),
+		Prefix: aws.String(key),
+	}
+
+	lg := a.log(ctx).WithFields(logging.Fields{
+		"qualified_ns":  qualifiedKey.GetStorageNamespace(),
+		"qualified_key": qualifiedKey.GetKey(),
+		"key":           obj.Identifier,
+	})
+
+	client := a.clients.Get(ctx, bucket)
+	resp, err := client.ListMultipartUploads(ctx, input)
+	if err != nil {
+		lg.WithError(err).Error("ListParts multipart uploads failed")
+		return nil, err
+	}
+
+	mpuResp := block.ListMultipartUploadsResponse{
+		Uploads: []types.MultipartUpload{},
+	}
+	mpuResp.Uploads = append(mpuResp.Uploads, resp.Uploads...)
```

Comment:
no need to initialize empty slice and append another - you can initialize the Uploads member with resp.Uploads.